cEP-0016.md

```diff
@@ -0,0 +1,234 @@
+# Git Commit Content Inspection
+
+| Metadata |                                                   |
+| -------- | ------------------------------------------------- |
+| cEP      | 16                                                |
+| Version  | 1.0                                               |
+| Title    | Git commit content inspection                     |
+| Authors  | Kriti Rohillla <mailto:kritirohilla567@gmail.com> |
+| Status   | Implementation due                                |
+| Type     | Process                                           |
+
+## Abstract
+
+This document describes coala process rules which will be enforced by the
+`GitCommitBear`.
+
+## Introduction
+
+The `GitCommitBear` at present performs a check on the content of regular
+commits made to coala. However, there are many special types of commit
+messages that should be used only in conjunction with patches containing a
+special type of content.
+
+For example, `git revert` commit which is used to revert any previous commit and
+`git merge` commit which is generated whenever someone performs a GitHub rebase
+and tries to merge master onto the pull request. Rebase and merge on GitHub will
+always update the committer information and create new commit SHAs which is not
+correct. These commits generate a default commit message that is very concise
+and not according to coala’s standards. The coala project wants to allow
+`git revert` commits, but does not want to accept `git merge` commits as the
+repositories are strictly fast-forward merges that do not have an extra commit.
+
+Also in cases where some tests are not required by any pull request made to
+coala, some special sequences can be added to git commit messages to instruct
+tools to operate in a certain way.
+
+This project is about detecting such commits and verifying that they are correct
+according to the configuration settings in `.coafile`
```

Comment:
Being unfamiliar and external to the project, this part is a bit confusing to me, it explains how several cases work, how you handle commits within the coala project… But it's unclear what is the problem you're trying to solve. When I reached the `Proposed Changes` section I headed back here, as I did not have a clear statement of what where the problems you're trying to solve.

In the  *For example* you explain what a revert and a merge are. Then you explain what rebase and merge done on github do. Then you talk about Coala project's standards. Then you talk about some *special sequences added to git commit messages* that do some stuff. Finally, it says that the project is about doing commit verification accordingly to a specification written in a `.coafile`.

But it's unclear to me what problem you're actually trying to solve with this project. And whether it's a problem within how coala projects work, or a generic problem for whoever uses the `GitCommitBear` tool.

So to help me understand what you're doing could you:

1. explain what the `GitCommitBear` tool does well (by explaining the current use cases)
2. what are the limitations of the `GitCommitBear` tool, and why it's a problem (by giving use cases)? (for coala projects, but also for any projects),
3. what do you want to change/improve to do so? (without going into the gory details, as you've got a section for that below 😉)